src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -4,22 +4,467 @@ Thomas Segismont
+
+=== Availability
+Represents the availability of a resource such as host machine (physical or virtual) or an application server. There
+are only three supported availability types or values:
+
+* up
+* down
+* unknown
+
+Availability is stored as single, unsigned byte.
+
+=== Gauge
+Has a numeric value that can fluctuate, going up or down. Some examples of gauges include,
+
+* Available heap space in the JVM
+* Number of active HTTP sessions on a web server
+* Disk space used by a database table
+* Execution time for a REST API call
+
+With each of these examples, values can increase or decrease. In some instances, like JVM heap space, there are
+well-defined bounds for the possible values.
+
+A gauge value is stored as a 64-bit floating point number.
+
+=== Counter
+Has a numeric value that monotonically increases or decreases. Some examples include:
+
+* Total number of requests to a REST endpoint
+* Total number of request timeouts for a Cassandra node
+* Total number of request timeouts for a Cassandra cluster
+
+These examples involve values that are always increase. Counters can also be decreasing.
+
+A counter value is stored as a 64-bit signed long.
+
+There are two types of counters commonly uses with time series databases (TSDB). One stores the current count or total with
+each data point. The other stores the delta or increment with each data point. The former is more commonly used with
+counters that can easily be maintained by the client. Tracking the total number of requests to a REST endpoint for a
+specific server can be done easily by the client. Tracking the total number of requests for the endpoint across all
+servers however is more challenging. This can be done more easily by storing the deltas and allowing the TSDB to
+compute and maintain the total count.
+
+Hawkular Metrics only supports the former in which each data point represents the total count; however, we can easily
+simulate counters that store deltas. See the <<Incrementing Counters>> section for more details.
+
+==== Rate
+A rate is a derived metric whose values are computed from counter data points. Rate data points can retrieved for any
+counter. They are stored as 64-bit floating point numbers.
+
+== Metric Ids
+Every metric has an id that uniquely identifies it. The id consists of three parts - the tenant id, the metric type,
+and the metric name. The tenant id is a variable length, UTF-8 encoded string. The metric type is stored as a one byte
+integer. The metric name is stored as a variable length, UTF-8 encoded string.
+
+The parts that comprise the metric id provide namespacing. A metric name only has to be unique for the metric type and
+the tenant. For example, suppose we have a tenant id of com.acme. The com.acme tenant could have a gauge named
+http_request_time and also have a counter named http_request_time.
+
+Just like with tenant ids, Hawkular Metrics does not perform any validation checks to ensure metric ids are unique.
```

Comment:
This is why registering the metricId before using it makes sense. To detect duplicates.